atlasdb-dbkvs/src/test/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/OracleDdlTableTest.java

```diff
@@ -231,19 +227,122 @@ public void dropTablesDoesNotThrowIfMultipleMatchingTableReferencesExistAndCaseS
         assertThatCode(() -> tableMappingDdlTable.drop()).doesNotThrowAnyException();
     }
 
+    @Test
+    public void createAltersTableIfConfiguredAndOverflowTableExists() throws TableMappingNotFoundException {
+        createTable();
+        createOverflowTable();
+        setTableToHaveOverflowColumn(false);
+        setMetadataToHaveOverflow(true);
+        assertThatCode(() -> tableMappingDdlTable.create(createMetadata(true))).doesNotThrowAnyException();
+        verifyNumberOfTimesTableAltered(1);
+    }
+
+    @Test
+    public void createDoesNothingWhenAlterSpecifiedButOverflowColumnExists() throws TableMappingNotFoundException {
+        createTable();
+        createOverflowTable();
+        setTableToHaveOverflowColumn(true);
+        setMetadataToHaveOverflow(true);
+        assertThatCode(() -> tableMappingDdlTable.create(createMetadata(true))).doesNotThrowAnyException();
+        verifyNumberOfTimesTableAltered(0);
+    }
+
+    @Test
+    public void createDoesNothingWhenAlterSpecifiedButMigrationUnstarted() throws TableMappingNotFoundException {
+        createTable();
+        createOverflowTable();
+        setTableToHaveOverflowColumn(false);
+        OracleDdlTable ddlTable = createOracleDdlTable(ImmutableOracleDdlConfig.builder()
+                .from(TABLE_MAPPING_DEFAULT_CONFIG)
+                .overflowMigrationState(OverflowMigrationState.UNSTARTED)
+                .build());
+        assertThatThrownBy(() -> ddlTable.create(createMetadata(true)))
+                .isInstanceOf(SafeIllegalArgumentException.class)
+                .hasMessageContaining(OracleDdlTable.MISSING_OVERFLOW_TABLE_MESSAGE);
+        verifyNumberOfTimesTableAltered(0);
+    }
+
+    @Test
+    public void createDoesNothingWhenAlterSpecifiedButOverflowColumnIsNotNeeded() throws TableMappingNotFoundException {
+        createTable();
+        createOverflowTable();
+        assertThatCode(() -> tableMappingDdlTable.create(createMetadata(false))).doesNotThrowAnyException();
+        verifyNumberOfTimesTableAltered(0);
+    }
+
+    @Test
+    public void createDoesNothingIfOverflowColumnNeededButTableNotListedInConfig()
+            throws TableMappingNotFoundException {
+        createTable();
+        createOverflowTable();
+        setTableToHaveOverflowColumn(false);
+        OracleDdlTable ddlTable = createOracleDdlTable(ImmutableOracleDdlConfig.builder()
+                .from(TABLE_MAPPING_DEFAULT_CONFIG)
+                .alterTablesOrMetadataToMatch(Set.of())
+                .build());
+        assertThatThrownBy(() -> ddlTable.create(createMetadata(true)))
+                .isInstanceOf(SafeIllegalArgumentException.class)
+                .hasMessageContaining(OracleDdlTable.MISSING_OVERFLOW_TABLE_MESSAGE);
+        verifyNumberOfTimesTableAltered(0);
+    }
+
     private void createTable() throws TableMappingNotFoundException {
+        // We do not always create the table, thus this is sometimes not used
```

Comment:
This doesn't map to the name :D What are the actual cases when this isn't used?

Reply:
lol ok yeah comment here is defo bad

Reply:
But the case I was trying to illustrate here, is that not every test for this class is calling `createTable(boolean overflow)`, and as a result, this mock is unused.

Reply:
Particularly your drop table tests!

Reply:
Ah, I see, it was referring to the OracleDdlTable#createTable method :)